### PR TITLE
simplify service definitions

### PR DESCRIPTION
**Pull request by wesol05**

### CODE REVIEW

File: app/config/services.yml [hunk 1]
```diff
@@ -1,37 +1,22 @@
...
     encoder.password:
         class: Symfony\Component\Security\Core\Encoder\BCryptPasswordEncoder
         arguments:
             - 12
 
-    provider.user_token_view:
-        class: AppBundle\Provider\UserTokenViewProvider
-        arguments:
-            - '@lexik_jwt_authentication.encoder'
+    AppBundle\Provider\UserTokenViewProvider: ~
```
Comment (matigda) [inline app/config/services.yml:22]: This is one of the cases why autowiring is not so great. We have JWTEncoderInterface injected into UserTokenViewProvider. Can You tell from the top of your head what exact service is being injected into UserTokenViewProvider? :>
The other thing is inconsistency in service definitions. Looking at the services.yml file we already have a a little mess, and we are halfway there. 
IMO it's good for RAD, but not for something that should be an example of good approach.
As Symfony documentation says:
"Service auto wiring was pioneered more than 10 years ago by the Spring Java framework with their @Autowired annotation and it's a feature best suited for simplifying the development of application prototypes."

Comment (wesol05) [reply app/config/services.yml:22]:
```
This is one of the cases why autowiring is not so great. We have JWTEncoderInterface injected into UserTokenViewProvider. Can You tell from the top of your head what exact service is being injected into UserTokenViewProvider? :>
```
For me this is actually a benefit, I don't need to know exact implementtion, only think I need to know is the behavior (interface) which can be easily found in class constructor

```
The other thing is inconsistency in service definitions. Looking at the services.yml file we already have a a little mess, and we are halfway there.
```
You have right, and that why we try to figure aout the best solution and then clean up it a little. This pr is proposition, it opens discussion ;)

```
Service auto wiring was pioneered more than 10 years ago by the Spring Java framework with their @Autowired annotation and it's a feature best suited for simplifying the development of application prototypes.
```
In this sentence there is no world saying not to use this. It speeds up aplication deelopment and simplify it, so for me these are pros of this solution

Comment (matigda) [reply app/config/services.yml:22]: "it's a feature best suited for simplifying the development of application prototypes" - I think we're not building an app prototype here :) 

And about being explicit in services.yml - when I go to services definitions I would like to know immediately what given service actual dependencies are. It's easier for debugging. You rather don't go to definition file just to see whether it's actually there, You are going there to get more details. If You don't want to know exact implementation then You are free to check the interface and You can get there from Your service class. No need to check services.yml ;)

**Comment by matigda** [comment]
As we agreed on priv, we will leave autowire for now.

**Pull request closed without merging**